### PR TITLE
 #74: Updated dependencies on top of 1.0.0

### LINKED ISSUE #74: Update dependencies on top of 1.0.0
We want to re-lock to resolve these vulnerable dependencies:

- https://github.com/exasol/error-reporting-python/security/dependabot/15 
- https://github.com/exasol/error-reporting-python/security/dependabot/17 
- https://github.com/exasol/error-reporting-python/security/dependabot/16 
- https://github.com/exasol/error-reporting-python/security/dependabot/18 

| Package          | Affected         | Patched |
|------------------|------------------|---------|
| urllib3 (pip)    | >= 1.22, < 2.6.3 | 2.6.3   |
| filelock (pip)   | < 3.20.3         | 3.20.3  |
| virtualenv (pip) | < 20.36.1        | 20.36.1 |
| pip (pip)        | < 26.0           | 26.0    |

Updates 
  - Updating urllib3 (2.3.0 -> 2.6.3)
  - Updating filelock (3.16.1 -> 3.20.3)
  - Updating virtualenv (20.29.0 -> 20.36.1)
  - Updating pip (24.3.1 -> 26.0.1)

### PR DESCRIPTION
**Pull request by ckunki**
Closes #74

### CODE REVIEW

File: .github/workflows/build-and-publish.yml [hunk 1]
```diff
@@ -4,21 +4,24 @@ on:
   workflow_call:
     secrets:
       PYPI_TOKEN:
-          required: true
```
Comment (ckunki) [inline .github/workflows/build-and-publish.yml:7]: All workflows are re-generated after updating to PTB version 5 without any manual change.

File: doc/changes/unreleased.md [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 
+## Refactoring
+
+* #72: Dropped support for Python 3.9 as EOL was 2025-10, which newer dependencies no longer support
 * #68: Updated to poetry 2.1.2 and exasol-toolbox 1.1.0
 * Updated transitive dependencies
 * #67: Extended unit test to contain test scenarios with different values for the parameter description
-* #72: Relocked transitive dependencies pip, urllib3, and filelock and updated exasol-toolbox to 4.0.0
\ No newline at end of file
+* #72: Relocked transitive dependencies pip, urllib3, and filelock and updated exasol-toolbox to 4.0.0
```
Comment (kratz00) [inline doc/changes/unreleased.md]:
```suggestion
* #72: Relocked transitive dependencies pip, urllib3, and filelock and updated exasol-toolbox to 5.0.0
```

Comment (kratz00) [reply doc/changes/unreleased.md]: Does not make sense, better mention the udate in a separate bullet point.

Comment (ckunki) [reply doc/changes/unreleased.md]: Ticket #72 included updating the PTB to version 4
Ticket #74 includes update to version 5.

**Review by kratz00 — DISMISSED** [review]

File: exasol/error/_parse.py [hunk 3]
```diff
@@ -263,38 +295,43 @@ def normalize(self, params: ast.Dict) -> Iterator[tuple[str, str]]:
...
+        Checks if keys of ast dictionary are of expected type.  If the
+        keys are of expected type, the method returns True, otherwise False.
         """
         ret_val = True
+        error_element = "parameter keys"
         for key in parameter_node.keys:
-            # The type of ast.Dict.keys is List[Optional[ast.expr]], not List[ast.expr] as someone would expect.
-            # However, trying unit tests with parameters of kind {None: "something"} did not provoke the "key" to be None.
-            # Nevertheless, keep the following error handling, in case of some strange corner case resulting "key" to be None.
+            # The type of ast.Dict.keys is List[Optional[ast.expr]], not
```
Comment (kratz00) [inline exasol/error/_parse.py]:
```suggestion
            # The type of ast.Dict.keys is List[ast.expr|None], not
```

File: exasol/error/_parse.py [hunk 4]
```diff
@@ -263,38 +295,43 @@ def normalize(self, params: ast.Dict) -> Iterator[tuple[str, str]]:
...
+            # List[ast.expr] as someone would expect.  However, trying unit
```
Comment (kratz00) [inline exasol/error/_parse.py]:
```suggestion
            # List[ast.expr] as someone would expect. However, trying unit
```

File: test/unit/cli_test.py [hunk 5]
```diff
@@ -325,93 +309,79 @@ def test_ErrorCollector_error_definitions(src, expected):
...
                     },
                 )
             ],
+            id="message",
         ),
+    ],
+)
+def test_error_definition_not_constant(src, expected):
+    """
+    Error definitions in python files may only use constant values.  This
```
Comment (kratz00) [inline test/unit/cli_test.py]:
```suggestion
    Error definitions in Python files may only use constant values. This
```

File: test/unit/cli_test.py [hunk 6]
```diff
@@ -325,93 +309,79 @@ def test_ErrorCollector_error_definitions(src, expected):
...
-            == first_expected._error._message_builder
-        )
-        assert first_error._error._mitigations == first_expected._error._mitigations
-        assert (
-            first_error._error._parameter_dict == first_expected._error._parameter_dict
-        )
+def test_value_not_string(src, expected) -> None:
+    """
+    Verify parsing errors in case a non-string value is provided for an
+    error attribute such as code, message, mitigation.
```
Comment (kratz00) [inline test/unit/cli_test.py]:
```suggestion
    error attribute such as code, message or mitigation.
```

**Review by kratz00 — DISMISSED** [review]

**Review by kratz00 — APPROVED** [review]
